### PR TITLE
feat: Show ticket wise analysis

### LINKED ISSUE #182: Show ticket wise analytics
Currently, only aggregate analysis of all tickets is done

We need to show ticket wise analysis along with the ticket type

test

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #182 

- Add circular progress ticket wise analysis
- Add ticket analyzer class

device-2017-06-30-014353

### CODE REVIEW

**Comment by mejariamol** [comment]
Looking real nice!
Just increase the padding. And why not showing ticket names?

**Comment by mejariamol** [comment]
Looking real nice!
Just increase the padding. And why not showing ticket names?

**Comment by iamareebjamal** [comment]
@mejariamol This is ticket type wise analysis, not every ticket is shown. Just the type only

**Comment by mejariamol** [comment]
How come two free tickets? Why are these shown separately.

**Comment by iamareebjamal** [comment]
There are 3 types of tickets in Open Event API. Paid, Free and Donation

**Comment by mejariamol** [comment]
>Paid, Free and Donation

Okay. I think you missed labeling `Donation` one ?

And increasing padding will look better.

**Comment by iamareebjamal** [comment]
Yeah. I missed that. Fixing the build and increasing padding

**Comment by iamareebjamal** [comment]
device-2017-06-30-043940

File: app/src/main/java/org/fossasia/openevent/app/event/detail/TicketAnalyser.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public static void analyseTotalTickets(@NonNull Event event) {
+        subscribeSingle(analyseTotalTicketsByType(event, TICKET_FREE), event.freeTickets);
+        subscribeSingle(analyseTotalTicketsByType(event, TICKET_PAID), event.paidTickets);
+        subscribeSingle(analyseTotalTicketsByType(event, TICKET_DONATION), event.donationTickets);
+    }
+
+    public static void analyseSoldTickets(@NonNull Event event, @NonNull List<Attendee> attendees) {
+        subscribeSingle(analyseSoldTicketsByType(attendees, TICKET_FREE), event.soldFreeTickets);
+        subscribeSingle(analyseSoldTicketsByType(attendees, TICKET_PAID), event.soldPaidTickets);
+        subscribeSingle(analyseSoldTicketsByType(attendees, TICKET_DONATION), event.soldDonationTickets);
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/detail/TicketAnalyser.java]: Actually no need do these 6 computations separately. Just one will be enough.
The approach will be:
keep all the observables you need to find
then for each attendee, increment concerned ones accordingly.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/detail/TicketAnalyser.java]: It's not a reactive approach and requires the result to be dependent on each other

**Pull request merged**